cms/djangoapps/contentstore/outlines_backfill.py

```diff
@@ -0,0 +1,138 @@
+"""
+Code which updates and backfills course outline data.
+
+Uses a proxy model to enable a Django admin interface to trigger asynch
+tasks which backfill/recreate course outline data.
+"""
+import logging
+from django.contrib import admin
+from django.contrib.admin.helpers import ACTION_CHECKBOX_NAME
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.content.learning_sequences.api import key_supports_outlines
+
+from .tasks import update_outline_from_modulestore_task, update_multiple_outlines_from_modulestore_task
+
+
+# If an admin user requests the course outline backfilling of this number of courses -or- higher,
+# create a celery task which creates a celery tasks to backfill each course outline.
+TASK_TO_LAUNCH_TASKS_THRESHOLD = 20
+
+log = logging.getLogger(__name__)
+
+
+class ReadOnlyAdminMixin(object):
```

Comment:
I wish I had thought to do this in `learning_sequences`! Might make a worthy addition to `edx-django-utils`.

Reply:
Agreed - I'll do this move in phase ][ of this work.